lua/lspconfig/util.lua

```diff
@@ -347,6 +359,7 @@ function M.server_per_root_dir_manager(make_config)
         return
       end
       attach_and_cache(root_dir, client_id)
+      register_client_single(vim.lsp.get_client_by_id(client_id))
```

Comment:
does this affect core lspconfig functionality, not only "lspinfo" ?

Reply:
It shouldn't have any side effects. When there is a file opened in single file mode. It create a  field called`single_buffers`  in client table. And insert this buffer. Clean up this field when the buffer is detach.

Reply:
but isn't it strange that LspInfo features require this kind of change? This is a hint that the `server_per_root_dir_manager` and all that "cache" code is doing too much. Why is lspconfig maintaining its own client cache...?

Reply:
we have a client cache but is simple cached the client id